osbuild/util/rhsm.py

```diff
@@ -0,0 +1,54 @@
+"""Red Hat Subscription Manager support module
+
+This module implements utilities that help with interactions
+with the subscriptions attached to the host machine.
+"""
+
+import configparser
+import re
+
+
+class Subscriptions:
+    def __init__(self, repositories):
+        self.repositories = repositories
+
+    @classmethod
+    def from_host_system(cls):
+        with open('/etc/yum.repos.d/redhat.repo', 'r') as fp:
+            return cls.parse_repo_file(fp)
+
+    @staticmethod
+    def _process_baseurl(input_url):
+        # First escape meta characters that might occur in a URL
+        input_url = re.escape(input_url)
+
+        # Now replace variables with regexes (see man 5 yum.conf for the list)
+        for variable in ['\\$releasever', '\\$arch', '\\$basearch', '\\$uuid']:
+            input_url = input_url.replace(variable, '.*')
```

Comment:
I am not the biggest master on regexes, so I summon @achilleas-k: I am just wondering about greedy / non-greedy matching of `.*`?